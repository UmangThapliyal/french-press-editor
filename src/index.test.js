import React from "react";

import { shallow } from "enzyme";

import { FrenchPress } from "./";

test("Render FrenchPress without crashing", () => {
  shallow(<FrenchPress />);
});
