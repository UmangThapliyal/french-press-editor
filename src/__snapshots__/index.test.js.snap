// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FrenchPress without crashing, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FrenchPress />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Unknown
        click={[Function]}
        style={
          Object {
            "display": "none",
            "opacity": "0",
            "top": 0,
          }
        }
      >
        <Unknown />
      </Unknown>
      <Editor
        autoCorrect={true}
        autoFocus={false}
        onChange={[Function]}
        onClick={[Function]}
        options={
          Object {
            "domain": "",
            "imagePlaceholder": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
          }
        }
        placeholder="Write your storyâ€¦"
        plugins={
          Array [
            Object {
              "onPaste": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onDrop": [Function],
              "onPaste": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
            Object {
              "onKeyDown": [Function],
            },
          ]
        }
        readOnly={false}
        renderMark={[Function]}
        renderNode={[Function]}
        schema={
          Object {
            "blocks": Object {
              "divider": Object {
                "isVoid": true,
              },
              "docket": Object {
                "isVoid": true,
              },
              "image": Object {
                "data": Object {
                  "src": [Function],
                },
                "isVoid": true,
              },
              "link": Object {
                "nodes": Array [
                  Object {
                    "objects": Array [
                      "text",
                    ],
                  },
                ],
              },
            },
            "document": Object {
              "last": Object {
                "types": Array [
                  "paragraph",
                ],
              },
              "nodes": Array [
                Object {
                  "types": Array [
                    "paragraph",
                    "heading",
                    "divider",
                    "quote",
                    "image",
                    "docket",
                    "link",
                  ],
                },
              ],
              "normalize": [Function],
            },
            "inlines": Object {
              "heading": Object {
                "nodes": Array [
                  Object {
                    "types": Array [
                      "text",
                    ],
                  },
                ],
              },
              "paragraph": Object {
                "nodes": Array [
                  Object {
                    "types": Array [
                      "text",
                      "link",
                    ],
                  },
                ],
              },
              "quote": Object {
                "nodes": Array [
                  Object {
                    "types": Array [
                      "text",
                    ],
                  },
                ],
              },
            },
          }
        }
        spellCheck={true}
        style={
          Object {
            "background": "",
            "boxShadow": "",
            "minHeight": "28em",
          }
        }
        value={
          Immutable.Record {
            "data": Immutable.Map {},
            "decorations": null,
            "document": Immutable.Record {
              "data": Immutable.Map {},
              "key": "1",
              "nodes": Immutable.List [
                Immutable.Record {
                  "data": Immutable.Map {},
                  "isVoid": false,
                  "key": "2",
                  "nodes": Immutable.List [
                    Immutable.Record {
                      "leaves": Immutable.List [
                        Immutable.Record {
                          "marks": Immutable.Set [],
                          "text": "",
                        },
                      ],
                      "key": "3",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
            },
            "history": Immutable.Record {
              "redos": Immutable.Stack [],
              "undos": Immutable.Stack [],
            },
            "schema": Immutable.Record {
              "stack": Immutable.Record {
                "plugins": Array [
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "schema": Object {},
                  },
                ],
              },
              "document": Object {
                "data": Object {},
                "nodes": null,
              },
              "blocks": Object {},
              "inlines": Object {},
            },
            "selection": Immutable.Record {
              "anchorKey": "3",
              "anchorOffset": 0,
              "focusKey": "3",
              "focusOffset": 0,
              "isBackward": false,
              "isFocused": false,
              "marks": null,
              "isAtomic": false,
            },
          }
        }
      />
      <input
        accept="image/png,image/jpeg"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
      />
    </div>,
    <Unknown
      formatCommand={[Function]}
      menuRef={[Function]}
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      value={
        Immutable.Record {
          "data": Immutable.Map {},
          "decorations": null,
          "document": Immutable.Record {
            "data": Immutable.Map {},
            "key": "1",
            "nodes": Immutable.List [
              Immutable.Record {
                "data": Immutable.Map {},
                "isVoid": false,
                "key": "2",
                "nodes": Immutable.List [
                  Immutable.Record {
                    "leaves": Immutable.List [
                      Immutable.Record {
                        "marks": Immutable.Set [],
                        "text": "",
                      },
                    ],
                    "key": "3",
                  },
                ],
                "type": "paragraph",
              },
            ],
          },
          "history": Immutable.Record {
            "redos": Immutable.Stack [],
            "undos": Immutable.Stack [],
          },
          "schema": Immutable.Record {
            "stack": Immutable.Record {
              "plugins": Array [
                Object {
                  "validateNode": [Function],
                },
                Object {
                  "validateNode": [Function],
                },
                Object {
                  "validateNode": [Function],
                },
                Object {
                  "validateNode": [Function],
                },
                Object {
                  "validateNode": [Function],
                },
                Object {
                  "validateNode": [Function],
                },
                Object {
                  "validateNode": [Function],
                },
                Object {
                  "validateNode": [Function],
                },
                Object {
                  "schema": Object {},
                },
              ],
            },
            "document": Object {
              "data": Object {},
              "nodes": null,
            },
            "blocks": Object {},
            "inlines": Object {},
          },
          "selection": Immutable.Record {
            "anchorKey": "3",
            "anchorOffset": 0,
            "focusKey": "3",
            "focusOffset": 0,
            "isBackward": false,
            "isFocused": false,
            "marks": null,
            "isAtomic": false,
          },
        }
      }
    />,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <Unknown
          click={[Function]}
          style={
            Object {
              "display": "none",
              "opacity": "0",
              "top": 0,
            }
          }
        >
          <Unknown />
        </Unknown>
        <Editor
          autoCorrect={true}
          autoFocus={false}
          onChange={[Function]}
          onClick={[Function]}
          options={
            Object {
              "domain": "",
              "imagePlaceholder": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
            }
          }
          placeholder="Write your storyâ€¦"
          plugins={
            Array [
              Object {
                "onPaste": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onDrop": [Function],
                "onPaste": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
              Object {
                "onKeyDown": [Function],
              },
            ]
          }
          readOnly={false}
          renderMark={[Function]}
          renderNode={[Function]}
          schema={
            Object {
              "blocks": Object {
                "divider": Object {
                  "isVoid": true,
                },
                "docket": Object {
                  "isVoid": true,
                },
                "image": Object {
                  "data": Object {
                    "src": [Function],
                  },
                  "isVoid": true,
                },
                "link": Object {
                  "nodes": Array [
                    Object {
                      "objects": Array [
                        "text",
                      ],
                    },
                  ],
                },
              },
              "document": Object {
                "last": Object {
                  "types": Array [
                    "paragraph",
                  ],
                },
                "nodes": Array [
                  Object {
                    "types": Array [
                      "paragraph",
                      "heading",
                      "divider",
                      "quote",
                      "image",
                      "docket",
                      "link",
                    ],
                  },
                ],
                "normalize": [Function],
              },
              "inlines": Object {
                "heading": Object {
                  "nodes": Array [
                    Object {
                      "types": Array [
                        "text",
                      ],
                    },
                  ],
                },
                "paragraph": Object {
                  "nodes": Array [
                    Object {
                      "types": Array [
                        "text",
                        "link",
                      ],
                    },
                  ],
                },
                "quote": Object {
                  "nodes": Array [
                    Object {
                      "types": Array [
                        "text",
                      ],
                    },
                  ],
                },
              },
            }
          }
          spellCheck={true}
          style={
            Object {
              "background": "",
              "boxShadow": "",
              "minHeight": "28em",
            }
          }
          value={
            Immutable.Record {
              "data": Immutable.Map {},
              "decorations": null,
              "document": Immutable.Record {
                "data": Immutable.Map {},
                "key": "1",
                "nodes": Immutable.List [
                  Immutable.Record {
                    "data": Immutable.Map {},
                    "isVoid": false,
                    "key": "2",
                    "nodes": Immutable.List [
                      Immutable.Record {
                        "leaves": Immutable.List [
                          Immutable.Record {
                            "marks": Immutable.Set [],
                            "text": "",
                          },
                        ],
                        "key": "3",
                      },
                    ],
                    "type": "paragraph",
                  },
                ],
              },
              "history": Immutable.Record {
                "redos": Immutable.Stack [],
                "undos": Immutable.Stack [],
              },
              "schema": Immutable.Record {
                "stack": Immutable.Record {
                  "plugins": Array [
                    Object {
                      "validateNode": [Function],
                    },
                    Object {
                      "validateNode": [Function],
                    },
                    Object {
                      "validateNode": [Function],
                    },
                    Object {
                      "validateNode": [Function],
                    },
                    Object {
                      "validateNode": [Function],
                    },
                    Object {
                      "validateNode": [Function],
                    },
                    Object {
                      "validateNode": [Function],
                    },
                    Object {
                      "validateNode": [Function],
                    },
                    Object {
                      "schema": Object {},
                    },
                  ],
                },
                "document": Object {
                  "data": Object {},
                  "nodes": null,
                },
                "blocks": Object {},
                "inlines": Object {},
              },
              "selection": Immutable.Record {
                "anchorKey": "3",
                "anchorOffset": 0,
                "focusKey": "3",
                "focusOffset": 0,
                "isBackward": false,
                "isFocused": false,
                "marks": null,
                "isAtomic": false,
              },
            }
          }
        />
        <input
          accept="image/png,image/jpeg"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
      </div>,
      <Unknown
        formatCommand={[Function]}
        menuRef={[Function]}
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        value={
          Immutable.Record {
            "data": Immutable.Map {},
            "decorations": null,
            "document": Immutable.Record {
              "data": Immutable.Map {},
              "key": "1",
              "nodes": Immutable.List [
                Immutable.Record {
                  "data": Immutable.Map {},
                  "isVoid": false,
                  "key": "2",
                  "nodes": Immutable.List [
                    Immutable.Record {
                      "leaves": Immutable.List [
                        Immutable.Record {
                          "marks": Immutable.Set [],
                          "text": "",
                        },
                      ],
                      "key": "3",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
            },
            "history": Immutable.Record {
              "redos": Immutable.Stack [],
              "undos": Immutable.Stack [],
            },
            "schema": Immutable.Record {
              "stack": Immutable.Record {
                "plugins": Array [
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "validateNode": [Function],
                  },
                  Object {
                    "schema": Object {},
                  },
                ],
              },
              "document": Object {
                "data": Object {},
                "nodes": null,
              },
              "blocks": Object {},
              "inlines": Object {},
            },
            "selection": Immutable.Record {
              "anchorKey": "3",
              "anchorOffset": 0,
              "focusKey": "3",
              "focusOffset": 0,
              "isBackward": false,
              "isFocused": false,
              "marks": null,
              "isAtomic": false,
            },
          }
        }
      />,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
